google-cloud-storage/lib/google/cloud/storage/bucket.rb

```diff
@@ -895,6 +895,63 @@ def policy_only_locked_at
           uniform_bucket_level_access_locked_at
         end
 
+        ##
+        # Whether the bucket's file IAM configuration enforces Public Access Prevention. The default is `false`. This
+        # value can be modified by calling {Bucket#public_access_prevention=}.
+        #
+        # @return [Boolean] Returns `false` if the bucket has no IAM configuration or if Public Access Prevention is
+        #   not `enforced` in the IAM configuration. Returns `true` if Public Access Prevention is `enforced` in the IAM
+        #   configuration.
+        #
+        # @example
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud::Storage.new
+        #
+        #   bucket = storage.bucket "my-bucket"
+        #
+        #   bucket.public_access_prevention = :enforced
+        #   bucket.public_access_prevention_enforced? # true
+        #
+        def public_access_prevention_enforced?
```

Comment:
If possible, we'd like to keep public access prevention as a string so that it mirrors the backend implementation.

Reply:
The reason for this is that if public access prevention is set to "unspecified" at the bucket level, it could still be inheriting an "enforced" value.

Reply:
Ok, I'll add the string read method and show that in the docs.

Reply:
Added `Bucket#public_access_prevention`, and replaced usages of `#public_access_prevention_enforced?` in samples  and docs with `#public_access_prevention`.

Reply:
@quartzmo Thanks for making this change! Could you please make it a string/enum in the library as well?

Reply:
Both strings and symbols are accepted. The library authors initially regarded symbols as more familiar to Rubyists, but I can change the samples for this feature to show strings: `bucket.public_access_prevention = "enforced"`. We don't have enums in the google-cloud-ruby libraries.

Reply:
I changed symbol arguments to strings in sample code and tests.